projects/plugins/jetpack/extensions/shared/components/media-player-control/index.js

```diff
@@ -0,0 +1,164 @@
+import { STATE_PAUSED, STATE_PLAYING, STORE_ID } from '../../../store/media-source/constants';
+import { convertSecondsToTimeCode } from './utils';
+
+function noop () {}
+
+export function MediaPlayerControl( {
+	skipForwardTime = 5,
+	jumpBackTime = 5,
+	syncMode,
+	onSyncModeToggle,
+	playIcon = 'controls-play',
+	pauseIcon = 'controls-pause',
+	jumpBackIcon = ControlBackFiveIcon,
+	skipForwardIcon = ControlForwardFiveIcon,
+	onTimeChange = noop,
+	currenTimeDisplay = true,
+} ) {
+	const {
+		playerState,
+		mediaCurrentTime,
+		mediaDuration,
+		defaultMediaSource,
+		mediaDomReference,
+	} = useSelect( select => {
+		const {
+			getMediaSourceCurrentTime,
+			getMediaPlayerState,
+			getDefaultMediaSource,
+			getMediaSourceDuration,
+			getMediaSourceDomReference,
+		} = select( STORE_ID );
+
+		return {
+			playerState: getMediaPlayerState(),
+			mediaCurrentTime: getMediaSourceCurrentTime(),
+			mediaDuration: getMediaSourceDuration(),
+			defaultMediaSource: getDefaultMediaSource(),
+			mediaDomReference: getMediaSourceDomReference(),
+		};
+	}, [] );
+
+	useEffect( () => {
+		if ( ! syncMode ) {
+			return;
+		}
+
+		if ( playerState !== STATE_PLAYING ) {
+			return;
+		}
+
+		onTimeChange( mediaCurrentTime );
+	}, [ mediaCurrentTime, onTimeChange, playerState, syncMode ] );
+
+	const timeInFormat = convertSecondsToTimeCode( mediaCurrentTime );
+	const isDisabled = ! defaultMediaSource;
+
+	const {
+		toggleMediaSource,
+		setMediaSourceCurrentTime,
+	} = useDispatch( STORE_ID );
+
+	function togglePlayer() {
+		toggleMediaSource( defaultMediaSource.id );
+	}
+
+	function setPlayerCurrentTime( time ) {
+		if ( mediaDomReference ) {
+			mediaDomReference.currentTime = time;
+		}
+		setMediaSourceCurrentTime( defaultMediaSource.id, time );
+	}
+
+	function setCurrentTime( time ) {
+		setPlayerCurrentTime( time );
+		if ( mediaDomReference ) {
+			mediaDomReference.currentTime = time;
+		}
+
+		if ( syncMode ) {
+			onTimeChange( time );
+		}
+	}
+
+	return (
+		<>
+			{ jumpBackTime !== false && (
+				<ToolbarButton
+					icon={ jumpBackIcon }
+					isDisabled={ isDisabled }
+					onClick={ () => setCurrentTime( mediaCurrentTime - jumpBackTime ) }
+					label={ __( 'Jump back', 'jetpack' ) }
+				/>
+			) }
+
+			<ToolbarButton
+				icon={ playerState === STATE_PAUSED ? playIcon : pauseIcon }
+				isDisabled={ isDisabled }
+				onClick={ togglePlayer }
+				label={ __( 'Play', 'jetpack' ) }
+			/>
+
+			{ skipForwardTime && (
+				<ToolbarButton
+					icon={ skipForwardIcon }
+					isDisabled={ isDisabled }
+					onClick={ () => setCurrentTime( mediaCurrentTime + skipForwardTime ) }
+					label={ __( 'Skip forward', 'jetpack' ) }
+				/>
+			) }
+
+			{ typeof syncMode !== 'undefined' && (
+				<ToolbarButton
+					icon={ syncMode ? ControlUnsyncIcon : ControlSyncIcon }
+					disabled={ isDisabled || ! mediaDuration }
+					onClick={ () => onSyncModeToggle( ! syncMode ) }
+					label={ __( 'Keep in-sync mode', 'jetpack' ) }
```

Comment:
Could the label for the sync button be "Sync timestamp"?

Reply:
https://github.com/Automattic/jetpack/pull/18390/commits/9035fe0fd09dd8e5bd743296be207ec4d8b8bf92